modules/escalation/main.tf

```
@@ -5,17 +5,17 @@ resource "opsgenie_escalation" "this" {
   description   = try(var.escalation.description, var.escalation.name)
   owner_team_id = try(var.escalation.owner_team_id, null)
 
-  rules {
-    condition   = try(var.escalation.rule.condition, "if-not-acked")
-    notify_type = try(var.escalation.rule.notify_type, "default")
-    delay       = try(var.escalation.rule.delay, 0)
+  dynamic "rules" {
+    for_each = try(var.escalation.rule, [])
```

Since we are supporting an array of rules lets make that clear with the variable name
```suggestion
    for_each = try(var.escalation.rules, [])
```